crates/ruff_cli/src/diagnostics.rs

```diff
@@ -474,14 +472,24 @@ pub(crate) fn lint_stdin(
         );
     }
 
+    let notebook_indexes = if let SourceKind::IpyNotebook(notebook) = source_kind {
+        FxHashMap::from_iter([(
+            path.map_or_else(|| "-".into(), |path| path.to_string_lossy().to_string()),
+            // Index needs to be computed always to store in cache.
```

Comment:
My understanding is that we don't perform any caching when using stdin.

Reply:
I think this is copied over from `lint_path`, but I don't fully understand it in the context of `lint_path`. Don't we need to compute the index to report diagnostics, not just store in the cache?

Reply:
I removed these comments as I found them sorta unhelpful.

Reply:
(I think this comment made sense in the PR in which it was introduced because it was a change, but it's less helpful in isolation.)